engine/execution/computation/manager.go

```diff
@@ -214,8 +215,19 @@ func (e *Manager) ComputeBlock(
 
 	e.programsCache.Set(block.ID(), toInsert)
 
-	if e.uploader != nil {
-		err = e.uploader.Upload(result)
+	if len(e.uploaders) > 0 {
+		var g errgroup.Group
+
+		for _, uploader := range e.uploaders {
+			uploader := uploader
+
+			g.Go(func() error {
+				return uploader.Upload(result)
+			})
+		}
+
+		err := g.Wait()
+
 		if err != nil {
 			return nil, fmt.Errorf("failed to upload block result: %w", err)
 		}
```

Comment:
Nit: may be useful to output information on which specific uploader failed here. Using `errgroup` obfuscates this information, so I'd suggest maybe not using it and instead manually implementing some logic to capture each individual error from each uploader and creating a `multierror` out of them.